bundles/binding/org.openhab.binding.isy/src/main/java/org/openhab/binding/isy/ISYActiveBinding.java

```diff
@@ -0,0 +1,718 @@
+	}
+
+	/**
+	 * @{inheritDoc
+	 */
+	@Override
+	protected void execute() {
+		// the frequently executed code (polling) goes here ...
+		if (this.logger.isDebugEnabled()) {
+			this.logger.debug("refreshing status");
+		}
+
+		queryStatus();
+	}
+
+	/**
+	 * @{inheritDoc
+	 */
+	@Override
+	protected void internalReceiveCommand(final String itemName,
+			final Command command) {
+
+		// the code being executed when a command was sent on the openHAB
+		// event bus goes here. This method is only called if one of the
+		// BindingProviders provide a binding for the given 'itemName'.
+		if (this.logger.isDebugEnabled()) {
+			this.logger.debug("internalReceiveCommand({},{}) is called!",
+					itemName, command);
+		}
+
+		ISYBindingConfig config = getBindingConfigByName(itemName);
+
+		if (config != null) {
+			processCommand(config, command);
+		} else {
+			this.logger.warn(
+					"No configured Insteon address found for item [{}]",
+					itemName);
+		}
+	}
+
+	/**
+	 * @{inheritDoc
+	 */
+	@Override
+	protected void internalReceiveUpdate(final String itemName,
+			final State newState) {
+		// the code being executed when a state was sent on the openHAB
+		// event bus goes here. This method is only called if one of the
+		// BindingProviders provide a binding for the given 'itemName'.
+		if (this.logger.isDebugEnabled()) {
+			this.logger.debug("internalReceiveUpdate({},{}) is called!",
+					itemName, newState);
+		}
+	}
+
+	/**
+	 * Callback from the ISY client when a node changed its state. Find the
+	 * corresponding openHAB item and send a notification on the event bus.
+	 * 
+	 * @param control
+	 *            {@link UDControl} ISY command
+	 * @param action
+	 *            Mostly a string with the value, e.g. "0", "255"
+	 * @param node
+	 *            {@link UDNode} ISY node can be a device or a group
+	 */
+	@Override
+	public void onModelChanged(final UDControl control, final Object action,
+			final UDNode node) {
+		Collection<ISYBindingConfig> configs;
+		State state = null;
+		ISYControl ctrl;
+
+		try {
+			ctrl = ISYControl.valueOf(control.name);
+		} catch (IllegalArgumentException ie) {
+			ctrl = ISYControl.UNDEFINED;
+		}
+
+		switch (ctrl) {
+		case ST:
+			// only propagate status updates
+
+			configs = getBindingConfigFromAddress(node.address, control.name);
+
+			for (ISYBindingConfig config : configs) {
+
+				switch (config.getType()) { 
+				case SWITCH:
+					if ("0".equals(action)) {
+						state = OnOffType.OFF;
+					} else {
+						state = OnOffType.ON;
+					}
+					break;
+				case GROUP:
+					if ("0".equals(action)) {
+						state = OnOffType.OFF;
+					} else {
+						state = OnOffType.ON;
+					}
+					break;
+				case CONTACT:
+					if ("0".equals(action)) {
+						state = OpenClosedType.CLOSED;
+					} else {
+						state = OpenClosedType.OPEN;
+					}
+					break;
+				case THERMOSTAT:
+					state = new DecimalType(
+							new DecimalType((String) action).doubleValue() * 0.5);
```

Comment:
You could use BigDecimal math to avoid imprecise results with native doubles.  Like 
```java
private static final BigDecimal HALF = new BigDecimal("0.5");
...
state = new DecimalType(new BigDecimal((String) action).multiply(HALF));
```